pkg/convert/json_ngxtranslate_test.go

```diff
@@ -0,0 +1,129 @@
+package convert
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.expect.digital/translate/pkg/model"
+)
+
+func Test_fromNgxTranslate(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name       string
+		serialized []byte
+		expected   model.Messages
+		wantErr    bool
+	}{
+		{
+			name:       "Not nested",
+			serialized: []byte(`{"message":"example"}`),
+			expected: model.Messages{
+				Messages: []model.Message{
+					{
+						ID:      "message",
+						Message: "example",
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:       "Nested normally",
+			serialized: []byte(`{"message":{"example":"message1"}}`),
+			expected: model.Messages{
+				Messages: []model.Message{
+					{
+						ID:      "message.example",
+						Message: "message1",
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:       "Nested with dot",
+			serialized: []byte(`{"message.example":"message1"}`),
+			expected: model.Messages{
+				Messages: []model.Message{
+					{
+						ID:      "message.example",
+						Message: "message1",
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:       "Nested mixed",
+			serialized: []byte(`{"message.example":"message1","msg":{"example":"message2"}}`),
+			expected: model.Messages{
+				Messages: []model.Message{
+					{
+						ID:      "message.example",
+						Message: "message1",
+					},
+					{
+						ID:      "msg.example",
+						Message: "message2",
+					},
+				},
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			result, err := FromNgxTranslate(tt.serialized)
+
+			assert.NoError(t, err)
```

Comment:
```suggestion
			if !assert.NoError(t, err) {
				return
			}
```
1 failed assertion instead of 2, when `err!=nil`